### ISSUE TITLE
Modal content suggestions

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon-website#3312]
Most of the challenges with the modal component have been opened as issues https://github.com/carbon-design-system/carbon/issue/12727 and discussions https://github.com/carbon-design-system/carbon/discussions/12726

However, there is additional material to do with the published guidance. Commented pdf exports of the component's [Usage](https://ibm.box.com/s/f7ymfis2ksn1ed3dx8usyj5uw2habgsw), [Style](https://ibm.box.com/s/p7jx28kg2rkfgeztlba83eyfig4rdgjj) pages should be examined, along with the [dialog pattern](https://ibm.box.com/s/ck39rkymdis2zwqhivawiwdzz5a5a8ij).

Comments surround inconsistency and repetition between the pages.
